pandas/core/reshape/encoding.py

```diff
@@ -236,10 +236,11 @@ def _get_dummies_1d(
     codes, levels = factorize_from_iterable(Series(data))
 
     if dtype is None:
-        dtype = np.dtype(np.uint8)
+        dtype = bool
     # error: Argument 1 to "dtype" has incompatible type "Union[ExtensionDtype, str,
     # dtype[Any], Type[object]]"; expected "Type[Any]"
-    dtype = np.dtype(dtype)  # type: ignore[arg-type]
+    else:
+        dtype = np.dtype(dtype)  # type: ignore[arg-type]
```

Comment:
Using `np.dtype(bool)` will spam deprecation warning.

> DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this 
warning, use  `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted 
the numpy scalar type, use `np.bool_` here.

Reply:
I'm not seeing that, which version of numpy gives that?
```
>>> import numpy as np
>>> np.dtype(bool)
dtype('bool')
>>> np.__version__
'1.23.3'
```

Reply:
1.23.3 produces it as well but not that way. `np.dtype(bool)` has no warnings but later somewhere in the code the warning is produced. Also if it gives you any clues, `np.zeros((4, 4), dtype=np.dtype(bool))` for example, doesn't produce warnings but `np.zeros((4, 4), dtype=np.bool)` does.

Reply:
do you know which test produced it? I tried making this change and running `pytest pandas/tests/reshape/test_get_dummies.py` and didn't get any warnings

I think the warning in `[pandas/tests/frame/methods/test_sort_values.py](https://github.com/pandas-dev/pandas/pull/48022/files/a5f709d07e78a54d31d167ea93371f01691d9bd1#diff-22ecc57d4d0b38dab48201ec7a07694b4ef7b97dd79633ec5d3add972d9117f8)` is more concerning though

Reply:
Yeah you're right. Not sure how the warning popped up!